dali/test/device_test.h

```diff
@@ -169,8 +170,8 @@ __device__ void suite_name##_##test_name##_body( \
  */
 #define DEVICE_TEST_CASE_BODY(suite_name, test_name, grid, block, ...) \
   dali::testing::TestStatus status; \
-  status.Init(); \
-  if (HasFailure()) return; \
+  if (!status.Init()) \
+    return; \
```

Comment:
The control flow going through HasFailure confused coverity - it though that failure to allocate the buffer may go unnoticed. Hopefully `ASSERT_EQ(...), false` will do the trick.